test/e2e/common/secrets.go

```diff
@@ -134,6 +137,79 @@ var _ = ginkgo.Describe("[sig-api-machinery] Secrets", func() {
 		secret, err := createEmptyKeySecretForTest(f)
 		framework.ExpectError(err, "created secret %q with empty key in namespace %q", secret.Name, f.Namespace.Name)
 	})
+
+	ginkgo.It("should patch a secret", func() {
+		ginkgo.By("creating a secret")
+
+		secretTestName := "test-secret-" + string(uuid.NewUUID())
+
+		// create a secret in the test namespace
+		_, err := f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Create(&v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretTestName,
+				Labels: map[string]string{
+					"testsecret-constant": "true",
+				},
+			},
+			Data: map[string][]byte{
+				"key": []byte("value"),
+			},
+			Type: "Opaque",
+		})
+		framework.ExpectNoError(err, "failed to create secret")
+
+		ginkgo.By("listing secrets in all namespaces to ensure that there are more than zero")
+		// list all secrets in all namespaces
+		secretsList, err := f.ClientSet.CoreV1().Secrets("").List(metav1.ListOptions{
+			LabelSelector: "testsecret-constant=true",
+		})
+		framework.ExpectNoError(err, "failed to list secrets")
+		framework.ExpectNotEqual(len(secretsList.Items), 0, "no secrets found")
+
+		foundCreatedSecret := false
+		var secretCreatedName string
+		for _, val := range secretsList.Items {
+			if val.ObjectMeta.Name == secretTestName && string(val.Data["key"]) == "value" {
+				foundCreatedSecret = true
+				secretCreatedName = val.ObjectMeta.Name
+			}
+		}
+		framework.ExpectEqual(foundCreatedSecret, true, "unable to find secret by its value")
+
+		ginkgo.By("patching the secret")
+		// patch the secret in the test namespace
+		secretPatchNewData := base64.StdEncoding.EncodeToString([]byte("value1"))
+		secretPatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{"testsecret": "true"},
+			},
+			"data": map[string][]byte{"key": []byte(secretPatchNewData)},
+		})
+		framework.ExpectNoError(err, "failed to marshal JSON")
+		_, err = f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Patch(secretCreatedName, types.StrategicMergePatchType, []byte(secretPatch))
+		framework.ExpectNoError(err, "failed to patch secret")
```

Comment:
IMO once we've patched the secret, we should get the secret back and confirm the patch has taken effect.  As written this test could be satisfied by an endpoint that does nothing but return 200 OK.  So here I would once again locate the secret by name/namespace (might want to pull out into a helper func), and then verify its data

Reply:
I agree with this idea, I've updated the test to include the check in 07e3c46.